core/cc/crash_handler.cpp

```diff
@@ -17,29 +17,28 @@
 #include "core/cc/crash_handler.h"
 #include "core/cc/log.h"
 
-namespace {
-auto handler = [] (const std::string& minidump_path, bool succeeded) {
+namespace core {
+
+CrashHandler::Handler CrashHandler::defaultHandler =
+        [] (const std::string& minidump_path, bool succeeded) {
             if (!succeeded) {
                 GAPID_ERROR("Failed to write minidump out to %s", minidump_path.c_str());
             }
-            return succeeded;
         };
-}
-
-namespace core {
-
-CrashHandler::HandlerFunction CrashHandler::defaultHandlerFunction = handler;
-
-void CrashHandler::setHandlerFunction(HandlerFunction newHandlerFunction) {
-    mHandlerFunction = newHandlerFunction;
-}
 
-void CrashHandler::unsetHandlerFunction() {
-    mHandlerFunction = defaultHandlerFunction;
+CrashHandler::Unregister CrashHandler::registerHandler(Handler handler) {
+    auto id = mNextHandlerID++;
+    mHandlers[id] = handler;
+    return [this, id] () {
+        mHandlers.erase(id);
+    };
 }
 
 bool CrashHandler::handleMinidump(const std::string& minidumpPath, bool succeeded) {
-    return mHandlerFunction(minidumpPath, succeeded);
+    for (const auto& it : mHandlers) {
+        it.second(minidumpPath, succeeded);
+    }
+    return true;
```

Comment:
At the very least this should be `return succeeded` since if the minidump was not created we should let any other error handling mechanisms go to work.
```
if (Debugger::isAttached()) {
    return false;
}
return succeeded;
```
Should allow a debugger to attach to the process on OS X without disabling the handler as well.

Reply:
> any other error handling mechanisms go to work

What other handling mechanisms?